ledfx/effects/gifplayer.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import os
+import time
+import urllib.request
+
+import voluptuous as vol
+from PIL import Image
+
+from ledfx.consts import LEDFX_ASSETS_PATH
+from ledfx.effects.twod import Twod
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class GifPlayer(Twod):
+    NAME = "GIF Player"
+    CATEGORY = "Matrix"
+    HIDDEN_KEYS = Twod.HIDDEN_KEYS + []
+    ADVANCED_KEYS = Twod.ADVANCED_KEYS + ["bilinear", "blur"]
+    DEFAULT_GIF_PATH = f"{os.path.join(LEDFX_ASSETS_PATH, 'animated.gif')}"
+
+    CONFIG_SCHEMA = vol.Schema(
+        {
+            vol.Optional(
+                "bilinear",
+                description="default NEAREST, use BILINEAR for smoother scaling, expensive on runtime takes a few ms",
+                default=False,
+            ): bool,
+            vol.Optional(
+                "url source", description="Load image from", default=""
+            ): str,
+            vol.Optional(
+                "bounce",
+                description="Bounce the GIF instead of looping",
+                default=False,
+            ): bool,
+            vol.Optional(
+                "GIF FPS", description="How fast to play the gif", default=10
+            ): vol.All(vol.Coerce(int), vol.Range(min=1, max=60)),
+        }
+    )
+
+    def __init__(self, ledfx, config):
+        super().__init__(ledfx, config)
+
+    def config_updated(self, config):
+        super().config_updated(config)
+        self.gif_fps = self._config["GIF FPS"]
+        self.bounce = self._config["bounce"]
+        self.resize = (
+            Image.BILINEAR if self._config["bilinear"] else Image.NEAREST
```

Comment:
as commented elsewhere I would just default to bilinear, as you are not doing this per render pass